DragaliaAPI/Features/SavefileUpdate/V13Update.cs

```diff
@@ -0,0 +1,42 @@
+using DragaliaAPI.Database.Repositories;
+using DragaliaAPI.Features.Emblem;
+using DragaliaAPI.Shared.Definitions.Enums;
+using DragaliaAPI.Shared.MasterAsset;
+
+namespace DragaliaAPI.Features.SavefileUpdate;
+
+public class V13Update : ISavefileUpdate
+{
+    private readonly IStoryRepository storyRepository;
+    private readonly IEmblemRepository emblemRepository;
+    private readonly ILogger<V13Update> logger;
+
+    public V13Update(
+        IStoryRepository storyRepository,
+        IEmblemRepository emblemRepository,
+        ILogger<V13Update> logger
+    )
+    {
+        this.storyRepository = storyRepository;
+        this.emblemRepository = emblemRepository;
+        this.logger = logger;
+    }
+
+    public int SavefileVersion => 13;
+
+    public async Task Apply()
+    {
+        int[] readStoryIdList = storyRepository.UnitStories.Where(x => (x.State == StoryState.Read) && x.StoryType == StoryTypes.Chara).Select(x => x.StoryId).ToArray();
+        int storyCharacterId;
+        int[] characterStoryList;
+        foreach (int readStoryId in readStoryIdList)
+        {
+            storyCharacterId = MasterAsset.UnitStory[readStoryId].ReleaseTriggerId;
+            characterStoryList = MasterAsset.CharaStories[storyCharacterId].storyIds;
+            if (readStoryId == characterStoryList.Last())
```

Comment:
```suggestion
            if (readStoryId == characterStoryList[-1])
```
Not sure if this is better for perf but it doesnt use the IEnumerable func